### ISSUE TITLE
Quality-related updates with tabular backend

### ISSUE THREAD

**Issue by jowr** [CoolProp/CoolProp#739]
Is the QT-update supposed to work with the tabular backends? 
```Python
from __future__ import print_function
import CoolProp
from CoolProp import AbstractState
from CoolProp.CoolProp import PropsSI

fluids = ["Water"]
rp = ["TTSE&REFPROP","BICUBIC&REFPROP","REFPROP"]
cp = ["TTSE&HEOS","BICUBIC&HEOS","HEOS"]

msgs=[]
for fluid in fluids:
    for factor in [1e-6,1e-4,1e-2]:
        out = []
        for backend in rp+cp:
            state         = AbstractState(backend,fluid)
            T_triple      = state.trivial_keyed_output(CoolProp.iT_triple)
            try: T_min    = state.trivial_keyed_output(CoolProp.iT_min)
            except: T_min = T_triple; msgs.append("Error for Tmin in "+backend)
            T_small = state.trivial_keyed_output(CoolProp.iT_critical)*factor
            state.update(CoolProp.QT_INPUTS, 0, max([T_triple,T_min])+T_small)
            T_str = "{0:5.1f}".format(state.keyed_output(CoolProp.iT))
            #out.append("{0:>5s}".format(Tstr))
            out.append(T_str)
        print(", ".join(out))

print("\n".join(msgs))
print("")

msgs=[]
for fluid in fluids:
    for factor in [1e-6,1e-4,1e-2]:
        out = []
        for backend in rp+cp:
            state         = AbstractState(backend,fluid)
            try: p_triple      = state.trivial_keyed_output(CoolProp.iP_triple)
            except: 
                T_triple = state.trivial_keyed_output(CoolProp.iT_triple) #PropSI("Ttriple","",0,"",0,backend.split("&")[-1]+"::"+fluid)
                p_triple = PropsSI("P","Q",0,"T",T_triple,backend.split("&")[-1]+"::"+fluid); msgs.append("Error for ptriple in "+backend)
            try: p_min    = state.trivial_keyed_output(CoolProp.iP_min)
            except: p_min = p_triple; msgs.append("Error for pmin in "+backend)
            p_small = state.trivial_keyed_output(CoolProp.iP_critical)*factor
            state.update(CoolProp.PQ_INPUTS, max([p_triple,p_min])+p_small, 0)
            T_str = "{0:5.1f}".format(state.keyed_output(CoolProp.iT))
            #out.append("{0:>5s}".format(Tstr))
            out.append(T_str)
        print(", ".join(out))

print("\n".join(msgs))
```
yields
```
  nan,   nan, 273.2,   nan,   nan, 273.2
  nan,   nan, 273.2,   nan,   nan, 273.2
  nan,   nan, 279.6,   nan,   nan, 279.6
Error for Tmin in TTSE&REFPROP
Error for Tmin in BICUBIC&REFPROP
Error for Tmin in TTSE&HEOS
Error for Tmin in BICUBIC&HEOS
Error for Tmin in TTSE&REFPROP
Error for Tmin in BICUBIC&REFPROP
Error for Tmin in TTSE&HEOS
Error for Tmin in BICUBIC&HEOS
Error for Tmin in TTSE&REFPROP
Error for Tmin in BICUBIC&REFPROP
Error for Tmin in TTSE&HEOS
Error for Tmin in BICUBIC&HEOS

  nan,   nan, 273.6, 273.6, 273.6, 273.6
  nan,   nan, 296.2, 296.2, 296.2, 296.2
  nan,   nan, 396.6, 396.6, 396.6, 396.6
Error for ptriple in TTSE&REFPROP
Error for pmin in TTSE&REFPROP
Error for ptriple in BICUBIC&REFPROP
Error for pmin in BICUBIC&REFPROP
Error for ptriple in REFPROP
Error for pmin in REFPROP
Error for ptriple in TTSE&REFPROP
Error for pmin in TTSE&REFPROP
Error for ptriple in BICUBIC&REFPROP
Error for pmin in BICUBIC&REFPROP
Error for ptriple in REFPROP
Error for pmin in REFPROP
Error for ptriple in TTSE&REFPROP
Error for pmin in TTSE&REFPROP
Error for ptriple in BICUBIC&REFPROP
Error for pmin in BICUBIC&REFPROP
Error for ptriple in REFPROP
Error for pmin in REFPROP
```
There are several problems here:
  - [x] Tmin is not implemented in the tabular backends
  - [x] the keyed output returns nan with throwing an error (QT and PQ updates)
  - [x] QT update does not work  in the tabular backends
  - [ ] PQ update does not work  in the tabular backends with REFPROP

**Comment by ibell** [comment]
I think the REFPROP is the only part unresolved as of now.

**Comment by ibell** [comment]
The PHI0dll function is causing problems

**Comment by jowr** [comment]
Great, thanks for the quick action. Let me know if I can help with anything.

**Comment by ibell** [comment]
Can you confirm that I have fixed all the bugs?  I think I have, and I need to pull the more-current version of REFPROP to compile on my machine.

**Comment by ibell** [comment]
Good news: I have REFPROP 9.11 compiling on my OS X development box.  I ran into some trouble, I'll file an issue
Bad news: REFPROP does not implement p_triple (as far as I can tell), should we do a flash call?  It'll be slow, and no longer a trivial output

**Comment by jowr** [comment]
What about calculating p_triple in the wrapper during instanciation? That would make it a trivial output for all subsequent calls.

**Comment by ibell** [comment]
OK, what I will do is calculate p_triple as the bubble point pressure connected with T_triple at instantiation, and cache it.  That should do it.

**Comment by ibell** [comment]
Fixed this one, current output:
```
273.2, 273.2, 273.2, 273.2, 273.2, 273.2
273.2, 273.2, 273.2, 273.2, 273.2, 273.2
279.6, 279.6, 279.6, 279.6, 279.6, 279.6


273.6, 273.6, 273.6, 273.6, 273.6, 273.6
296.2, 296.2, 296.2, 296.2, 296.2, 296.2
396.6, 396.6, 396.6, 396.6, 396.6, 396.6
```

**Comment by ibell** [comment]
closed by 261e61f03e06aaa9fee07001221ffebae0d4db70
